lib/AgentManager.js

```diff
@@ -0,0 +1,414 @@
+// AgentManager: A class that manages the device agent.
+// It can start/stop and reload the agent
+
+const { initLogger, info, warn, debug } = require('./log')
+const { default: got } = require('got/dist/source')
+const agent = require('./agent')
+const os = require('os')
+const path = require('path')
+const fsExtra = require('fs-extra')
```

Comment:
What operation requires `fs-extra`? I find Node 16+ core `fs` has pretty much everything I ever resorted to `fs-extra` for.

Reply:
Using async / promises, I could devise no simple means of determining if a file exists (depreciated in fs) and 

Happy to revise is necessary.

Reply:
`fs.existsSync` works just fine and isn't deprecated. Require it from the non-promise fs module...

```
const { existsSync } = require ('fs')
```

Reply:
I'm not getting notifications of comments! Will revise now.

Reply:
Resolved in [1fefb9f](https://github.com/flowforge/flowforge-device-agent/pull/56/commits/1fefb9f1eb71469e62c17b4f851a6705fcc56cf1)